flexbox.md

```diff
@@ -0,0 +1,273 @@
+
+#### You Tell Me: What Should I Try?
+
+```html
+<body>
+  <div>This is my div!</div>
+</body>
+```
+
+```CSS
+html {
+  height: 100%;
+}
+
+body {
+  min-height: 100%;
+  background-color: #ccc;
+  margin: 0 auto;
+}
+
+div {
+  width: 100px;
+  height: 100px;
+  outline: 1px solid red;
+}
+```
+
+<details>
+  <summary><strong>These might work...</strong></summary>
+
+  > **Padding**: The simplest approach would be to set equal padding on the top and bottom of the element. This can get tedious, however, when there is more than one element in a container.
+  >
+  > **Absolute Positioning**: You could use properties like `top` and `left` to position an element in the center. This, however, removes it from the document flow.
+
+</details>
+
+<details>
+  <summary><strong>These could work in other scenarios...</strong></summary>
+
+  > **`line-height`**: When vertically centering a single line of text, you can set the line-height to that of the whole container.
+  >
+  > **`vertical-align`**: Used to align words within a line of text (e.g., superscript, subscript).
+  >
+  > **`display: table`**: ???
+
+</details>
+
+#### An Aside
+
+Vertical alignment has been the laughingstock of CSS for years. How can something so obvious be so difficult to accomplish?
+
+> HTML was created as a document-oriented language. CSS emerged as a way to make an HTML file appear more document-like. Literally, like something you would make in Microsoft Word.
+>
+> So layout wasn't much of a concern in the beginning. But as the web has evolved, so has the needs of developers. So why hasn't a standard for vertical alignment emerged accordingly?
+>
+> It's difficult to establish new CSS standards. The [CSS Working Group](https://en.wikipedia.org/wiki/CSS_Working_Group) has a hard time agreeing on anything, especially cross-browser standards.
+
+Fortunately, Flexbox has slowly but surely become a standard over the past few years...
+
+### Flexbox to the Rescue
+
+```CSS
+html {
+  height: 100%;
+}
+
+body {
+  min-height: 100%;
+  background-color: #ccc;
+  margin: 0 auto;
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+}
+
+div {
+  width: 100px;
+  height: 100px;
+  outline: 1px solid red;
+}
+```
+
+## How It Works (10 minutes / 0:10)
+
+![flexbox diagram](img/flexbox-diagram.jpg)
+
+When you declare `display: flex` on a container, it becomes a **flex container**.
+
+First, you use `flex-direction` to indicate whether you want the items in the container -- the **flex items** -- to "read" left-to-right (`row`), right-to-left (`row-reverse`), top-to-bottom (`column`), **or** bottom-to-top (`column-reverse`).
+
+When you specify a flex-direction, you can think of it as placing an axis in that direction across your flex container. So if you use `flex-direction: row` or `row-reverse`, this **main axis** will be the same as the X-axis (horizontal) on a graph. Otherwise, it'll be the Y-axis.
```

Comment:
diagram might be nice here

![](https://cdn.css-tricks.com/wp-content/uploads/2011/08/flexbox.png)